features/FEATURE_BLE/ble/BLE.h

```diff
@@ -54,109 +196,151 @@ class BLE
     };
 
     /**
-     * Callback type used by the onEventsToProcess function.
+     * Events to process event handler
+     */
+    typedef FunctionPointerWithContext<OnEventsToProcessCallbackContext*>
+        OnEventsToProcessCallback_t;
+
+    /**
+     * Register a callback called when the BLE stack has pending work.
+     *
+     * By registering a callback, application code can know when event processing
+     * has to be scheduled.
+     *
+     * @param on_event_cb Callback invoked when there are new events to process.
      */
-    typedef FunctionPointerWithContext<OnEventsToProcessCallbackContext*> OnEventsToProcessCallback_t;
+    void onEventsToProcess(const OnEventsToProcessCallback_t& on_event_cb);
 
     /**
-     * The context provided to init-completion-callbacks (see init() below).
+     * Process ALL pending events living in the BLE stack and return once all
+     * events have been consumed.
      *
-     * @param  ble
-     *             A reference to the BLE instance being initialized.
-     * @param  error
-     *             Captures the result of initialization. It is set to
-     *             BLE_ERROR_NONE if initialization completed successfully. Else
-     *             the error value is implementation specific.
+     * @note: this function is automatically called by the OS on mbed OS 3
```

Comment:
Query: Should we stop referring to Mbed 3? Also, aren't Mbed OS classic and Mbed OS 2 the same thing?

Reply:
Yes I will amend part of the documentation referencing mbed OS 3, it made sense at the time, but not anymore.